magenta/models/music_vae/lstm_models.py

```diff
@@ -247,24 +247,39 @@ def _flat_reconstruction_loss(self, flat_x_target, flat_rnn_output):
 
     Args:
       flat_x_target: The flattened ground truth vectors, sized
-        [sum(x_length), self._output_depth].
+        `[sum(x_length), self._output_depth]`.
       flat_rnn_output: The flattened output from all timeputs of the RNN,
-        sized [sum(x_length), rnn_output_size].
+        sized `[sum(x_length), rnn_output_size]`.
     Returns:
-      r_loss: The unreduced reconstruction losses, sized [sum(x_length)].
+      r_loss: The unreduced reconstruction losses, sized `[sum(x_length)]`.
       metric_map: A map of metric names to tuples, each of which contain the
         pair of (value_tensor, update_op) from a tf.metrics streaming metric.
       truths: Ground truth labels.
       predictions: Predicted labels.
     """
     pass
 
-  def _decode(self, batch_size, helper, z, max_length=None, x_input=None):
+  def _decode(self, z, helper, max_length=None, x_input=None):
+    """Decodes the given batch of latent vectors vectors, which may be 0-length.
+
+    Args:
+      z: Batch of latent vectors, sized `[batch_size, z_size]`, where `z_size`
+        may be 0 for unconditioned decoding.
+      helper: A seq2seq.Helper to use. If a TrainingHelper is passed and a
+        CudnnLSTM has previously been defined, it will be used instead.
+      max_length: (Optinal) The maximum iterations to decode.
+      x_input: (Optional) The inputs to the decoder for teacher forcing.
+        Required if CudnnLSTM is to be used.
+
+    Returns:
+      final_output: The final seq2seq.BasicDecoderOutput.
+    """
     initial_state = initial_cell_state_from_embedding(
-        self._dec_cell, z, batch_size, name='decoder/z_to_initial_state')
+        self._dec_cell, z, name='decoder/z_to_initial_state')
 
-    if (isinstance(helper, seq2seq.TrainingHelper) and
-        self._cudnn_dec_lstm):
+    # CudnnLSTM is only a replacement for TrainingHelper since it does not
+    # support sampling.
+    if type(helper) == seq2seq.TrainingHelper and self._cudnn_dec_lstm:  # pylint:disable=unidiomatic-typecheck
```

Comment:
would it be helpful to raise an exception if cudnn_dec_lstm = True and scheduled sampling is on?